retrieval/discovery/ec2.go

```diff
@@ -110,12 +115,25 @@ func (ed *EC2Discovery) refresh() (*config.TargetGroup, error) {
 				labels := model.LabelSet{
 					ec2LabelInstanceID: model.LabelValue(*inst.InstanceId),
 				}
+				labels[ec2LabelPrivateIP] = model.LabelValue(*inst.PrivateIpAddress)
+				addr := fmt.Sprintf("%s:%d", *inst.PrivateIpAddress, ed.port)
+
 				if inst.PublicIpAddress != nil {
 					labels[ec2LabelPublicIP] = model.LabelValue(*inst.PublicIpAddress)
+					labels[ec2LabelPublicDns] = model.LabelValue(*inst.PublicDnsName)
+					// replace address with public dns name if present
+					addr = fmt.Sprintf("%s:%d", *inst.PublicDnsName, ed.port)
```

Comment:
In that case, without a publicIp the privateIp would be used. Anyway I understand the point, I will drop that and use relabeling. I guess this can be re-evaluated in case other users will have the same scenario.